test/unit/StreamrClient.test.js

```diff
@@ -321,22 +323,36 @@ describe('StreamrClient', () => {
         describe('BroadcastMessage', () => {
             beforeEach(() => client.connect())
 
-            it('should call the message handler of each subscription', () => {
+            it('should call the message handler of each subscription', (done) => {
                 connection.expect(new SubscribeRequest('stream1'))
 
                 const counter = sinon.stub()
+                counter.onFirstCall().returns(1)
+                counter.onSecondCall().returns(2)
 
                 client.subscribe({
                     stream: 'stream1',
-                }, counter)
+                }, () => {
```

Comment:
There must be a simpler way! I can't write it off the top of my head, but I was thinking something like returning a promise that resolves on the second call. Thought about how to do it with sinon stubs easily, but I didn't figure it out immediately. Moving on.